test/kino_db/connection_cell_test.exs

```diff
@@ -181,16 +181,36 @@ defmodule KinoDB.ConnectionCellTest do
                  Req.new(http_errors: :raise)
                  |> ReqAthena.attach(
                    access_key_id: "id",
-                   secret_access_key: "secret",
-                   region: "region",
                    database: "default",
-                   output_location: "s3://my-bucket"
+                   output_location: "s3://my-bucket",
+                   region: "region",
+                   secret_access_key: "secret",
+                   token: "",
+                   workgroup: ""
                  )
 
                :ok\
                """
     end
 
+    test "doesn't restore source code with empty conditional fields" do
+      attrs = %{
+        "variable" => "db",
+        "type" => "athena",
+        "access_key_id" => "id",
+        "secret_access_key" => "secret",
+        "region" => "region",
+        "database" => "default",
+        "token" => "token",
+        "workgroup" => "",
+        "output_location" => ""
+      }
+
+      {_kino, source} = start_smart_cell!(ConnectionCell, attrs)
```

Comment:
I think in this case we can test just `to_source`, as we do in the other cell. Similarly for other tests. For actual restore what matters is whether the cell can start properly given no attrs or some attrs. wdyt?

Reply:
So, you mean to test the conditional and required fields?

Reply:
I mean that we can switch the "restore" tests (other than the first two) into `to_source` tests. The first two test the initialization behaviour with/without params, and in the other tests we are mostly concerned with the exact source code. Does that make sense?

Reply:
Yeah, I'll push some changes about that